### PR TITLE
feat(feature-verification): [skp-189] verification feature added

### PR DESCRIPTION
**Pull request by doncevas**

### CODE REVIEW

File: apps/ppm/src/app/app.tsx [hunk 1]
```diff
@@ -33,17 +31,19 @@ export const App = () => {
     <div className="app">
       <BrowserRouter basename="/">
         <Switch>
+          {console.log(PrivateRouter)
```
Comment (Neutronas) [inline apps/ppm/src/app/app.tsx:34]: Remove console.log

File: libs/data-access/http-requests/src/lib/data-access-http-requests.ts [hunk 2]
```diff
@@ -34,6 +36,17 @@ export const registration = (data: RegistrationInterface) => {
   return response;
 };
 
+export const verify = ( code: string) => {
```
Comment (Neutronas) [inline libs/data-access/http-requests/src/lib/data-access-http-requests.ts:39]: (code: string)

File: libs/data-access/authorization/src/lib/types.ts [hunk 3]
```diff
@@ -5,9 +5,11 @@ import { ApplicationRootState } from 'types';
 /* --- STATE --- */
 
 interface AuthState {
-  readonly user: any;
-  readonly isLoggedIn: boolean;
+  readonly user?: any;
+  readonly isLoggedIn?: boolean;
   readonly loading: boolean;
+  readonly registrationStep?: number;
+
```
Comment (Neutronas) [inline libs/data-access/authorization/src/lib/types.ts:12]: no empty line needed

File: apps/api/src/app/models/users/users.service.ts [hunk 4]
```diff
@@ -51,6 +52,7 @@ export class UsersService {
 
   isValidEmail (email : string){
     if(email){
+      // eslint-disable-next-line no-useless-escape
```
Comment (Gardemarin) [inline apps/api/src/app/models/users/users.service.ts:55]: Remove this comment

File: libs/features/registration/src/lib/features-registration.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      width: 100%;
+      height: 100%;
+      position: relative;
+    }
+
+    &__spinner {
+      position: absolute;
+      left: 50%;
+      top: 50%;
+      // transform: translate(-50%, -50%);
```
Comment (Gardemarin) [inline libs/features/registration/src/lib/features-registration.scss:20]: Remove this line

File: libs/features/registration/src/lib/features-registration.tsx [hunk 6]
```diff
@@ -1,21 +1,81 @@
...
+  const verifyEmail = (code: string) => {
+    dispatch(authorizationActions.verification(code));
+  };
+
+  const clearRegistrationStateData = () => {
+    clearRegistrationData();    
+    window.location.href = PrivateRoutesPath.POST_REGISTRY;
+  }
+
+  const getStepContent = (stepIndex) => {
```
Comment (Gardemarin) [inline libs/features/registration/src/lib/features-registration.tsx:45]: add type to `stepIndex`

File: libs/features/registration/src/lib/features-registration.tsx [hunk 7]
```diff
@@ -1,21 +1,81 @@
...
+function getSteps() {
+  return ['Registration', 'Verification', 'Login'];
+}
+
+const stateSelector = createStructuredSelector({
+  registrationStep: authorizationSelectors.selectRegistrationStep(),
+  loading: authorizationSelectors.selectLoading()
+});
 
+export const FeaturesRegistration = (props) => {
```
Comment (Gardemarin) [inline libs/features/registration/src/lib/features-registration.tsx:28]: Add type to `props`

**Pull request merged**